src/scheduler/server_info.c

```diff
@@ -223,8 +223,12 @@ query_server(status *pol, int pbs_sd)
 		return NULL;
 	}
 
+	sched = pbs_statsched(pbs_sd, NULL, NULL);
+	while(sched && strncmp(PBS_DFLT_SCHED_NAME, sched->name, PBS_MAXSCHEDNAME)) {
+		sched = sched->next;
```

Comment:
Same as above, check sched != NULL.  Use strcmp() and use a for() loop.

Reply:
1. while (ss) == while (ss != NULL) . I prefer former succinct format.
2. why let strcmp() run amok if by chance we get a non-null termintated string. I say its safe to restrict it to "n"
3. Ok I will change that, but I dont think there will be any difference in the compiled binary code